lib/util/support.js

```diff
@@ -10,6 +10,28 @@ const SUPPORTED_CHROME_BASED_BROWSERS = [
   'headlesschrome'
 ];
 
+/**
+ * Get the top level parenthesized substrings within a given string.
+ * Ex: "abc (def) gh(ij) (kl (mn)o)" => [""(def)", "(ij)", "(kl (mn)o)"]
+ * @param {string} string
+ * @returns {string[]}
+ */
+function getParenthesizedSubstrings(string) {
+  const openParenthesisPositions = [];
+  const substrings = [];
+  for (let i = 0; i < string.length; i++) {
+    if (string[i] === '(') {
+      openParenthesisPositions.push(i);
+    } else if (string[i] === ')' && openParenthesisPositions.length > 0) {
+      const openParenthesisPosition = openParenthesisPositions.pop();
+      if (openParenthesisPositions.length === 0) {
+        substrings.push(string.substring(openParenthesisPosition, i + 1));
+      }
+    }
```

Comment:
Looks like this code is forgiving if an extra close parenthesis is encountered. like  "(foo))(hello)"
Looks like it will return "(foo)" and "(hello)" which is not bad, but a commenting it in else case here would make it clear.

Reply:
@makarandp0 I will add a line in the documentation for the function. 👍